core/src/main/java/com/linecorp/armeria/client/ResponseTimeoutMode.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+/**
+ * Specifies when to start scheduling a response timeout. Note that this value does not affect calculation
+ * of the response timeout, but determines when the response timeout will affect the ongoing request.
```

Comment:
> Note that this value does not affect calculation of the response timeout

Question) What advantage can users expect when `REQUEST_SENT` is used? If the time of `decorators(3s)---connection acquisition(2s)` are always taken into account, `REQUEST_START` seems always better and supersedes other options.

Reply:
I imagine if there are users who are using a short `responseTimeout`, then requests made will still send the request over the wrire before failing.
I'm unsure if defaulting to `REQUEST_START` is a breaking change that is worth making.

Reply:
We may need a consensus on that before proceeding with it. Let's listen to other folk's opinions because the current behavior is likely to be controversial.

Reply:
I realized there was a mistake in the javadocs 😅    Pushed a commit to address this 👍